### ISSUE TITLE
Test Status in fileScreenings overview

### ISSUE THREAD

**Issue by nicolasvanermen** [ices-tools-dev/esas#92]
We figured this out: the appartent mingling was due to an old file (ESAS_BE_DATA_test_obis_tab_ERID.csv) that was in the database, instead of the most recent upload (ESAS_BE_DATA_test_obis_tab.cs), both with the same campaignID; very glad this was figured out, and that there is a logical explanation... Still very confusing because in the overview of file screenings it seems to suggest that the latter file (88) was uploaded to the database, hereby overwriting the former file (86):

![image](https://user-images.githubusercontent.com/7381853/174751958-67bfa859-807d-4759-a184-81f295cb91f5.png)

- [ ] So concluding: it remains to be tested whether the status "File already uploaded to the database" actually holds true.

_Originally posted by @nicolasvanermen in https://github.com/ices-tools-dev/esas/issues/88#issuecomment-1161421056_

**Comment by cmspinto** [comment]
> The appartent mingling in the downloaded Observations table was due to an old file (ESAS_BE_DATA_test_obis_tab_ERID.csv) that was in the database, instead of the most recent upload (ESAS_BE_DATA_test_obis_tab.cs), both with the same campaignID. Very confusing because in the overview of file screenings it seems to suggest that the latter file (88) was uploaded to the database, hereby overwriting the former file (86):
> 
> ![image](https://user-images.githubusercontent.com/7381853/174751958-67bfa859-807d-4759-a184-81f295cb91f5.png)
> 
>     * [ ]  So concluding: it remains to be tested whether the status "File already uploaded to the database" actually holds true.

Files were uploaded to the database, but deleted after!

**Comment by nicolasvanermen** [comment]
OK, this could explain this but hard for me to check. The only confusing thing remaining was the reported submission date, that was set on 17/06 while the false file (screening 86) was pushed to the DB at 8/06. 

- [ ] To be tested!

**Comment by nicolasvanermen** [comment]
@cmspinto: comparable problem now, see my file screening 133. File was submitted (resubmission of file with screening 130), file status says that file has been uploaded, but it is not in the download... The other resubmission (screening 132) did make it in the download. What am I missing?

**Comment by cmspinto** [comment]
Nicolas, 

You are the DataRightHolder to organization 202 but not for Organization 630.
![image](https://user-images.githubusercontent.com/8705477/174788746-19c98499-8120-4636-9b2f-5c5a03c7a022.png)

I have not implemented yet what we talked in the morning. I need to look carefully before changing the upload permission logic. Does this make sense?

**Comment by nicolasvanermen** [comment]
OK Carlos, but the problem is that the list of file screenings does not notify me that there was no actual resubmission. It simply says: "File already uploaded to the database". This is also the subject of this issue.
Did you change it now, because I tried it again an now both resubmissions are in the download...??
![image](https://user-images.githubusercontent.com/7381853/174799274-2c11b889-8431-459c-ba1d-f0f8dacf3c05.png)

**Comment by nicolasvanermen** [comment]
SO to be clear: I screen a file, it says there are no warnings, it says it is a resubmission, I opt to resubmit, it says data are downloaded to the database, in the fileScreenings overview it says the file has been uploaded... Only: the file is not in the database!

**Comment by cmspinto** [comment]
Yes, have changed it now to always allow a user to overwrite their own files!

**Comment by nicolasvanermen** [comment]
Both resubmissions are now in the list of the submissions; just a few questions:

- [ ] clicking on the numbers gives an empty page, what is supposed to be there? (@cmspinto )
- [ ] files that are resubmitted are deleted from this list, is that OK? (@peterdesmet)
- [ ] why do the submission ID's jump from 131 to 135 to 140, I did not delete any file after the meeting? (@cmspinto )

**Comment by cmspinto** [comment]
> * why do the submission ID's jump from 131 to 135 to 140, I did not delete any file after the meeting? (@cmspinto )

Because I tested some files and deleted them immediately after!

**Comment by nicolasvanermen** [comment]
[ESAS_BE_DATA_test_resubmission_Carlos.csv](https://github.com/ices-tools-dev/esas/files/8949008/ESAS_BE_DATA_test_resubmission_Carlos.csv)
Or could you upload this file? 

- [ ] DRH NIOZ, uploaded by Carlos, so I shouldn't be able to overwrite it, and the application should warn me thereof

**Comment by cmspinto** [comment]
Done, I can see both: 
![image](https://user-images.githubusercontent.com/8705477/174804624-320a1978-73cb-4e6d-a255-7ec6ad627901.png)

I'm 630 DataRightHolder, so I can see, overwrite or delete your file.

**Comment by nicolasvanermen** [comment]
![image](https://user-images.githubusercontent.com/7381853/174810627-2a7f42fc-f966-45ab-bcd7-bf9758aad45b.png)

**Comment by nicolasvanermen** [comment]
![image](https://user-images.githubusercontent.com/7381853/174810719-394ffa64-18d6-46d6-97f4-4e18d11d9792.png)

**Comment by nicolasvanermen** [comment]
Cleared this out during chat with Carlos: in case the file includes campaigns for which the submitter has no rights to overwrite them, there will be a warning and the submitter will not be able to upload the file (even if this file also includes brand new campaignID's!).

- [ ] test warning pop-up

**Comment by nicolasvanermen** [comment]
OK I tested this as well and it worked fine. Would be nice if the warning simply stated:

- [ ] "You do not have permission to overwrite the campaign(s) in the database"

**Comment by cmspinto** [comment]
![image](https://user-images.githubusercontent.com/8705477/175488805-1907b889-f1ac-4feb-a8d6-5320d55dadd5.png)
